### PR TITLE
Drop operand.namespace

### LINKED ISSUE #114: operand.namespace should be dropped
**What happened**:

I create nfd CR:

```
kubectl apply -f - << EOF
apiVersion: nfd.kubernetes.io/v1
kind: NodeFeatureDiscovery
metadata:
  name: nfd-master-server
  namespace: operators
spec:
  operand:
    image: 'k8s.gcr.io/nfd/node-feature-discovery:v0.10.1'
    namespace: nfd
EOF
```

NFD get's deployed in operators namespace, as it should. The `operand.namespace` field has no effect (as it shouldn't as it shouldn't be allowed to "contaminate" arbitrary namespaces with NFD) and it should be dropped from the CRD.

**What you expected to happen**:

The `operand.namespace` parameter should not be accepted at al.

**How to reproduce it (as minimally and precisely as possible)**:

See above.

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Fixes #114

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
> @ArangoGutierrez: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> pull-node-feature-discovery-operator-verify	[73bfac9](https://github.com/kubernetes-sigs/node-feature-discovery-operator/commit/73bfac9aa9b4fbd94e8b3e675dfb4cb34101eece)	[link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_node-feature-discovery-operator/115/pull-node-feature-discovery-operator-verify/1490782952546635776)	true	`/test pull-node-feature-discovery-operator-verify`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes-sigs&repo=node-feature-discovery-operator&pr=115). [Your PR dashboard](https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3AArangoGutierrez). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes-sigs/node-feature-discovery-operator/issues?q=is:issue+is:open) when you hit one in your PR.

ups.... topology things got into the PR jejeje

**Review by marquiz — CHANGES_REQUESTED** [review]
The commits don't seem to be consistent as the first commit already drops operand.namespace. Either squash the commits into one or only do  `make manifests` in the first commit.

Also, commit messages (that do not have a "topic: " prefix) should begin with a capital letter.

File: controllers/nodefeaturediscovery_controller.go [hunk 1]
```diff
@@ -157,49 +157,49 @@ func (r *NodeFeatureDiscoveryReconciler) Reconcile(ctx context.Context, req ctrl
 	}
 
 	// Check the status of the NFD Operator Worker ServiceAccount
-	if rstatus, err := r.getWorkerServiceAccountConditions(ctx); err != nil {
+	if rstatus, err := r.getWorkerServiceAccountConditions(ctx, instance); err != nil {
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controller.go:160]: Hmm, this PR seems to bring other changes, too. Like supporting `instance` config parameter(?)

File: main.go [hunk 2]
```diff
@@ -82,6 +83,12 @@ func main() {
 		os.Exit(0)
 	}
 
+	watchNamespace, err := utils.GetWatchNamespace()
+	if err != nil {
+		klog.Error(err, "unable to get WatchNamespace, "+
+			"the manager will watch and manage resources in all namespaces")
+	}
```
Comment (marquiz) [inline main.go:90]: Woudln't it be ok to watch for all namespaces by default?

File: controllers/nodefeaturediscovery_controller.go (hunk 1 above)
Comment (ArangoGutierrez) [reply controllers/nodefeaturediscovery_controller.go:160]: https://github.com/ArangoGutierrez/node-feature-discovery-operator/blob/devel/v1spec_ns/controllers/nodefeaturediscovery_status.go#L497

Comment (ArangoGutierrez) [reply controllers/nodefeaturediscovery_controller.go:160]: before it was empty

File: main.go (hunk 2 above)
Comment (ArangoGutierrez) [reply main.go:90]: if https://github.com/ArangoGutierrez/node-feature-discovery-operator/blob/devel/v1spec_ns/pkg/utils/utils.go#L28 is not defined, if will default to all, so is ok.

File: controllers/nodefeaturediscovery_controller.go (hunk 1 above)
Comment (marquiz) [reply controllers/nodefeaturediscovery_controller.go:160]: Haha, sorry I got totally confused by the `instance` in this PR 🙈

File: docs/get-started/quick-start.md [hunk 3]
```diff
@@ -25,6 +25,11 @@ IMAGE_TAG={{ site.container_image }}
 make deploy
 ```
 
+By default the operator will watch the namespace defined on the
+deployment manifest the env WATCH_NAMESPACE will track the
+metadata.namespace of the deployment, if unset the operator
+will watch ALL namespaces.
```
Comment (marquiz) [inline docs/get-started/quick-start.md:31]: Hmm, that wording was a bit cryptic to me 🧐 

```suggestion
By default the operator will watch `NodeFeatureDiscovery` objects
only in the namespace where the operator is deployed in. This is
specified by the `WATCH_NAMESPACE` env variable in the operator
deployment manifest. If the  unset the operator will watch ALL
namespaces.
```

File: docs/get-started/quick-start.md [hunk 4]
```diff
@@ -25,6 +25,12 @@ IMAGE_TAG={{ site.container_image }}
 make deploy
 ```
 
+By default the operator will watch `NodeFeatureDiscovery` objects
+only in the namespace where the operator is deployed in. This is
+specified by the `WATCH_NAMESPACE` env variable in the operator
+deployment manifest. If the  unset the operator will watch ALL
```
Comment (marquiz) [inline docs/get-started/quick-start.md:31]: Puching myself 🤦 

```suggestion
deployment manifest. If unset the operator will watch ALL
```

**Review by marquiz — APPROVED** [review]
Looks ready to get in 👍 
/lgtm

**Pull request merged**